scripts/reconcile_dh_index/main.go

```diff
@@ -0,0 +1,240 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+
+	finderhttpclient "github.com/ipni/storetheindex/api/v0/finder/client/http"
+	"github.com/ipni/storetheindex/api/v0/finder/model"
+	"github.com/libp2p/go-libp2p/core/peer"
+	b58 "github.com/mr-tron/base58/base58"
+	"github.com/multiformats/go-multihash"
+)
+
+// This program can be used to reconcile results from regular and double hashed clients.
+// It works as following:
+// * fetch list of providers from source and target indexers
+// * find providers that exist in both indexers and that have the same last ad cid
+// * for each provider fetch last add via provider CLI
```

Comment:
It would be really nice if we can make this script to not depend on `provider` binary for ease of use, specially considering storetheindex offers go libraries to do what the provider CLI would do and this script lives in storetheindex repo.